### PR TITLE
Update documentation on parameter dictionaries with lists

### LINKED ISSUE #1813: Include conversion in nest.Create() API documentation
This Issue is a reminder to include the conversion suggested in #1534 in the API documentation of ``nest.Create()``.

### PR DESCRIPTION
**Pull request by stinebuu**
I have updated the documentation about parameter dictionaries with lists when creating nodes, or when setting parameters with `SetStatus()` and `set()`. Docstrings are updated, the neuron tutorial is updated a little, and I have added the information in the NEST 2 to NEST 3 transistion guide.

This fixes #1813

### CODE REVIEW

File: doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst [hunk 1]
```diff
@@ -337,6 +337,25 @@ or a ``nest.Parameter``
 Note that some parameters, like `global_id`, cannot be set. The documentation of a specific model
 will point out which parameters can be set and which are read-only.
 
+
+Dictionary with lists when setting parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (sarakonradi) [inline doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst:342]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

File: doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst [hunk 2]
```diff
@@ -337,6 +337,25 @@ or a ``nest.Parameter``
...
+It is now possible to use a dictionary with lists when setting node parameters
+with ``Create()``, ``set()`` or ``SetStatus()``. The values of the lists will 
+be distributed across the nodes. The way to do this previously was to apply a
+lists of dictionaries. Note that this is still possible.
```
Comment (sarakonradi) [inline doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst:346]:
```suggestion
list of dictionaries. This is still possible.
```

File: doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst [hunk 3]
```diff
@@ -337,6 +337,25 @@ or a ``nest.Parameter``
...
+
+The values in the single dictionary can also be single values, the value will
```
Comment (sarakonradi) [inline doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst:348]:
```suggestion
The values in the single dictionary can also be single values; the value will
```

File: doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst [hunk 4]
```diff
@@ -337,6 +337,25 @@ or a ``nest.Parameter``
...
+then be applied to each node. You can mix and match as you want, the dictionary
```
Comment (sarakonradi) [inline doc/guides/nest2_to_nest3/nest2_to_nest3_overview.rst:349]:
```suggestion
then be applied to each node. You can mix and match as you want; the dictionary
```

File: doc/tutorials/pynest_tutorial/part_2_populations_of_neurons.rst [hunk 5]
```diff
@@ -109,13 +109,16 @@ populations and will also be returned by the function ``Models()``.
 It is also possible to create populations with an inhomogeneous set of
 parameters. You would typically create the complete set of parameters,
 depending on experimental constraints, and then create all the neurons
-in one go. To do this supply a dictionaries with lists of the same length
-as the number of neurons (or synapses) created:
+in one go. To do this supply a dictionary with lists the same length
```
Comment (sarakonradi) [inline doc/tutorials/pynest_tutorial/part_2_populations_of_neurons.rst:112]:
```suggestion
in one go. To do this, supply a dictionary with lists the same length
```

File: doc/tutorials/pynest_tutorial/part_2_populations_of_neurons.rst [hunk 6]
```diff
@@ -109,13 +109,16 @@ populations and will also be returned by the function ``Models()``.
...
+as the number of neurons (or synapses) created. The dictionary can also
+contain single values, these will then be applied to each node.
```
Comment (sarakonradi) [inline doc/tutorials/pynest_tutorial/part_2_populations_of_neurons.rst:114]:
```suggestion
contain single values, which will then be applied to each node.
```

**Review by sarakonradi — CHANGES_REQUESTED** [review]
Thanks for addressing #1813! Here are a few minor suggestions.

**Review by terhorstd — APPROVED** [review]
LGTM :+1:

**Review by sarakonradi — APPROVED** [review]
Thanks!

**Pull request merged**